### DISCUSSION TITLE
How do I configure Homebrew with custom CLI path?

### DISCUSSION THREAD

**Discussion by andre4ik3** [Homebrew/discussions#655] [Getting started] [answered]
I am running Homebrew in an environment where I don't have admin.

- Homebrew is located in `~/homebrew`.
- CLT is installed in `~/bin`, `~/include`, `~/lib`, `~/libexec` and is in PATH. Compiling works fine.
- I can install bottles manually by getting the URL from the API, then by placing all the files in their respective places. They work perfectly fine, at least the ones that I use.
- However, even with `--force-bottle`, Homebrew refuses to install a bottle (complains about prefix not being default), and opts to build from source, which fails, because "Command Line Tools are not installed."
- `xcode-select --install` results in "Unable to install the software."
- Because of this, I literally can't install anything via `brew`.

The question: how do I configure Homebrew so that it installs from bottles despite the prefix and / or detects CLI in my custom installation?

**Comment by SMillerDev** [comment]
https://docs.brew.sh/Installation#untar-anywhere

Why not try this install method instead of trying to force a custom directory structure? Because you'll have to do a lot of patching to make that work.

**Reply by andre4ik3** [reply]
That's exactly what I've done. I've untar-d into `~/homebrew` but I literally cannot install anything because of the problems described above (Xcode CLT refusing to install normally, I installed in a custom directory, everything else builds fine, Brew won't even try building and just exit instantly)

**Comment by carlocab** [comment, marked as answer]
The CLT path is hardcoded into `brew`. You'll have to look in https://github.com/Homebrew/brew/tree/master/Library/Homebrew/os/mac, for example, to see how to change it. I'd start in `sdk.rb` and `xcode.rb`.

I'd try doing a user-only installation of the Xcode app first. There's a possibility `brew` will be able to recognise that and use it. Otherwise, you'll need to start modifying the source.

**Reply by arumoy** [reply]
Would it be acceptable to make the `PKG_PATH` overridable with another variable of sorts, e.g. `HOMEBREW_PKG_PATH` or `HOMEBREW_CLT_PATH`? I'd like to try and do that in that case.

I have my CLT in `~/Library/Developer/CommandLineTools`, and can't update python in brew anymore.
